ERCS/erc-7969.md

```diff
@@ -0,0 +1,261 @@
+---
+eip: 7969
+title: DKIM Registry Interface
+description: An interface for registering and validating DKIM public key hashes onchain.
+author: Mike Fu (@fumeng00mike), Matthew Yu (@0xknon), Ernesto García (@ernestognw)
+discussions-to: https://ethereum-magicians.org/t/new-erc-dkim-registry-interface/24530
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-06-11
+---
+
+## Abstract
+
+This EIP proposes a standard interface for registering and validating DKIM (DomainKeys Identified Mail) public key hashes on the Ethereum blockchain. The interface allows domain owners to register their DKIM public key hashes and enables third parties to verify the validity of these hashes.
+
+## Motivation
+
+With the growing adoption of Account Abstraction [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337) and the emergence of ZK Email Technology, there is a need for a standardized way to verify email ownership on-chain. This EIP provides a crucial building block for these technologies by enabling the verification of DKIM signatures through on-chain registries.
+
+This standard enables several important use cases:
+
+1. **Account Abstraction**: When combined with zkEmail, this registry enables email-based account abstraction. Users can prove ownership of their email address through DKIM signatures, allowing them to:
+   - Create and manage smart contract wallets
+   - Sign transactions using their email credentials
+   - Implement social recovery mechanisms
+2. **Account Recovery**: The registry facilitates secure account recovery mechanisms:
+   - Users can recover their wallet access by on-chain proving email ownership
+   - The process is trustless and secure due to the cryptographic nature of DKIM
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Interface
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code within the document itself must be CC0-1.0 licensed. We do permit MIT code in the `assets/` directory and you can link to it there.